src/core/plugin.js

```diff
@@ -0,0 +1,59 @@
+const {ChildProcess} = require("child_process");
+const treekill = require("treekill");
+const {execute, checkServer} = require("../core/executor");
+const Locale = require("../locales")("service.run");
+
+module.exports = function ({command, config}, processor = {}) {
+  if (!processor.BeforeAll || !processor.AfterAll) {
+    throw new Error(
+      "Please provide a processor containing the methods:  BeforeAll, AfterAll"
+    );
+  }
+
+  processor.Before(async function (scenario) {
+    // In order to fetch the dynamic data from extenral sources.
+    // Let's parse the scenario to get all the placeholder
+    await this.data.parse(scenario);
+  });
+
+  processor.Before({tags: "@skip or @wip"}, function () {
+    this.skipped = true;
+    return "skipped";
+  });
+
+  if (command) {
+    let server;
+    processor.BeforeAll(async function () {
+      if (typeof command === "string") {
+        const { mock } = this.restqa || {};
+        const { http } = mock || {}
+
+        server = await execute(command, http);
+        const restqapi = getPluginConfig("@restqa/restqapi", config);
+        if (!restqapi) {
+          throw new Error(Locale.get("error_missing_restqapi"));
+        }
+        const {url} = restqapi;
+        if (!url) {
+          throw new Error(Locale.get("error_missing_url"));
+        }
```

Comment:
Maybe should we make these check before executing: `server = await execute(command, http);`? WDYT?